src/jwks.cc

```diff
@@ -184,16 +182,24 @@ Status extractJwkFromJwkEC(const ::google::protobuf::Struct& jwk_pb,
     return Status::JwksECKeyBadCrv;
   }
   jwk->crv_ = crv_str;
+
+  // If both alg and crv specified, make sure they match
+  if (jwk->alg_specified_ && !jwk->crv_.empty()) {
+    if (!((jwk->alg_ == "ES256" && jwk->crv_ == "P-256") ||
+          (jwk->alg_ == "ES384" && jwk->crv_ == "P-384") ||
+          (jwk->alg_ == "ES512" && jwk->crv_ == "P-521"))) {
+      return Status::JwksECKeyAlgNotCompatibleWithCrv;
+    }
+  }
+
   int nid;
-  if (jwk->alg_ == "ES256" && (jwk->crv_ == "P-256" || jwk->crv_ == "")) {
+  if (jwk->alg_ == "ES256" || jwk->crv_ == "P-256") {
```

Comment:
is the `crv` required in the JWK? The behavior changes if both `alg` and `crv` is not set after this PR, is this expected?

Reply:
Added the default handling, not to break existing customers.